controllers/templatesync/template_sync.go

```diff
@@ -866,3 +1043,67 @@ func getLatestStatusMessage(pol *policiesv1.Policy, tIndex int) string {
 
 	return tmplDetails.History[0].Message
 }
+
+func hasClusterwideFinalizer(pol *policiesv1.Policy) bool {
+	for _, finalizer := range pol.Finalizers {
+		if finalizer == utils.ClusterwideFinalizer {
+			return true
+		}
+	}
+
+	return false
+}
+
+// removeFinalizer iterates over the finalizers and removes the one specified from the policy.
+func removeFinalizer(pol *policiesv1.Policy, finalizer string) {
+	for idx, fzr := range pol.Finalizers {
+		if fzr == finalizer {
+			pol.Finalizers = append(pol.Finalizers[:idx], pol.Finalizers[idx+1:]...)
```

Comment:
You either need to break after this or handle this differently since you are ranging over a slice that you are modifying in the range, which isn't safe.